src/analyzer/analyzerscheduledtrack.cpp

```diff
@@ -0,0 +1,16 @@
+#include "analyzer/analyzerscheduledtrack.h"
+
+#include "analyzer/analyzertrack.h"
+#include "track/trackid.h"
+
+AnalyzerScheduledTrack::AnalyzerScheduledTrack(TrackId trackId, AnalyzerTrack::Options options)
+        : trackId(trackId), options(options) {
+}
+
+const TrackId AnalyzerScheduledTrack::getTrackId() const {
```

Comment:
I did intend to return by value here since I understood `TrackId` to be a small, trivially copyable type. You are right that `const` doesn't make much sense in that case. Do you think it would be better to return by ref here?

Reply:
I think `sizeof(TrackId)` = 4 so that can be safely returned by value. On the other hand a receiving copy constructor takes always a const reference, so that this should be the preferred method of returning member variables. 

I would go for 
```suggestion
const TrackId& AnalyzerScheduledTrack::getTrackId() const {
```